core/src/main/java/de/medavis/lct/core/patcher/BomPatcher.java

```diff
@@ -0,0 +1,256 @@
+        componentMetaDataManager = new ComponentMetaDataManager();
+    }
+
+    private void init() {
+        configuration
+                .getSpdxLicensesUrl()
+                .ifPresentOrElse(
+                        url -> spdxLicenseManager.load(URI.create(url.toString())),
+                        spdxLicenseManager::loadDefaults
+                );
+
+        configuration
+                .getComponentMetadataUrl()
+                .map(url -> URI.create(url.toString()))
+                .ifPresent(componentMetaDataManager::load);
+
+        componentMetaDataManager.validateLicenseMappedNames(spdxLicenseManager.getSupportedLicenseNames());
+    }
+
+    /**
+     * Patches a BOM file.
+     *
+     * @param sourceFile Source BOM file
+     * @param targetFile Target BOM file
+     * @return Returns true if patching was successful. False, when BOM leaved untouched.
+     */
+    public boolean patch(@NotNull Path sourceFile, @NotNull Path targetFile) {
+        try (InputStream in = Files.newInputStream(sourceFile)) {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+            boolean result = patch(in, out);
+
+            LOGGER.info("Writing patched file '{}'", targetFile);
+            Files.createDirectories(targetFile.getParent());
+            Files.writeString(targetFile, out.toString(StandardCharsets.UTF_8), StandardOpenOption.CREATE);
+
+            return result;
+        } catch (IOException ex) {
+            throw new LicensePatcherException(ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * Patches a BOM stream.
+     *
+     * @param in  Source BOM as input stream
+     * @param out Target BOM as output stream
+     * @return Returns true if patching was successful. False, when BOM leaved untouched.
+     */
+    public boolean patch(@NotNull InputStream in, @NotNull OutputStream out) {
+        init();
+
+        try {
+            final byte[] bomBytes = ByteStreams.toByteArray(in);
+            Bom bom = BomParserFactory.createParser(bomBytes).parse(bomBytes);
+
+            Version version = Arrays
+                    .stream(Version.values())
+                    .filter(v -> v.getVersionString().equals(bom.getSpecVersion()))
+                    .findFirst()
+                    .orElseThrow(() -> new LicensePatcherException("Unsupported version: " + bom.getSpecVersion()));
+
+            if (!"CycloneDX".equals(bom.getBomFormat())) {
+                throw new LicensePatcherException("Unsupported BOM format: " + bom.getBomFormat());
+            }
+
+            String originalBom = BomGeneratorFactory.createJson(version, bom).toJsonString();
+
+            Optional.ofNullable(bom.getComponents())
+                    .orElse(List.of())
+                    .forEach(component -> {
+                        String purl = Objects.toString(component.getPurl(), "");
+
+                        if (component.getLicenses() == null) {
```

Comment:
The code is pretty hard to understand. I am not sure if I understand that right:
- If there are no licenses in the BOM, write the ones from component meta data into the BOM
-  If there are licenses in the BOM, check if they conform with the SPDX list, but otherwise ignore them and do the same as in the case above.

Is that correct? It does not really make sense to me... Additionally, it only takes into account the component meta data, but not the license mappings.
What would make sense to me would be the following algorithm:
- Transform component from BOM into `ComponentMetaData, enriching or replacing the licenses from those configured in component metadata, license mappings and license list. This logic is already implemented in `de.medavis.lct.core.list.ComponentLister#enrichWithMetadata` and could be reused.
- Write the licenses from ComponentMetadata into the BOM component